portal-impl/src/com/liferay/portal/service/impl/OrganizationLocalServiceImpl.java

```diff
@@ -976,6 +977,10 @@ public int getOrganizationsAndUsersCount(
 			companyId, parentOrganizationId, queryDefinition);
 	}
 
+	public List<Organization> getOrganizationsByLogoId(long logoId) {
+		return OrganizationUtil.findByLogoId(logoId);
```

Comment:
I don't get this... I mean... it's the only method that call the OrganizationUtil.*

Reply:
Hi @smottal ,

The stable errors were due to the the call not being from the service layer. We apparently can't call the finder method directly from WebServerServlet, so I created this as a workaround. The alternative would be to create a Transaction Executor manually in WebServerServlet, but I didn't think that was preferable over this. I agree that it looks weird, but it was the most straightforward way I could figure out how to use the Finder method like BChan wants although it's very much possible I'm accidentally overlooking a better alternative solution.

Reply:
but why this cannot call the finder in the persistence layer? like in line 899